plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/AbstractXmlValidatorTest.java

```diff
@@ -29,58 +33,99 @@
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Path;
+import org.eclipse.jst.common.project.facet.core.JavaFacet;
+import org.eclipse.jst.j2ee.web.project.facet.WebFacetUtils;
+import org.eclipse.wst.common.project.facet.core.IProjectFacetVersion;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+import org.eclipse.wst.validation.ValidationFramework;
+import org.eclipse.wst.validation.Validator;
 import org.junit.After;
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
 
 public class AbstractXmlValidatorTest {
 
-  private final String ELEMENT_MESSAGE = "Project ID should be specified at deploy time.";
-  private static IResource resource;
-  private static IProject project;
+  private static final IProjectFacetVersion APPENGINE_STANDARD_FACET_VERSION_1 =
+      ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID).getVersion("1");;
+  private IFile resource;
 
-  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator();
+  @ClassRule public static TestProjectCreator appEngineStandardProjectCreator =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+          WebFacetUtils.WEB_25, APPENGINE_STANDARD_FACET_VERSION_1));
+  
+  @ClassRule public static TestProjectCreator dynamicWebProjectCreator =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+                                                                    WebFacetUtils.WEB_25));
 
-  @BeforeClass
-  public static void setUp() throws CoreException {
-    project = projectCreator.getProject();
+  @Before
+  public void setUp() throws CoreException {
+    IProject project = dynamicWebProjectCreator.getProject();
     createFolders(project, new Path("src/main/webapp/WEB-INF"));
```

Comment:
So just be aware that new Dynamic Web Projects actually put the DD in `WebContent/WEB-INF/web.xml`, not `src/main/webapp/WEB-INF/web.xml`.  It doesn't matter for your validator as you're applying to all web.xml files, but maybe the tests should either make this explicit with a comment?